Libraries/PeriphDrivers/Source/WDT/wdt_ai87.c

```diff
@@ -157,3 +157,38 @@ void MXC_WDT_ClearIntFlag(mxc_wdt_regs_t *wdt)
 {
     MXC_WDT_RevB_ClearIntFlag((mxc_wdt_revb_regs_t *)wdt);
 }
+
+int MXC_WDT_SetClockSource(mxc_wdt_regs_t *wdt, mxc_wdt_clock_t clock_source)
+{
+    const uint8_t clock_source_num = 8;
+    uint8_t idx = 0;
+    uint8_t instance = 0;
+    mxc_wdt_clock_t clock_sources[2][8] = {
```

Comment:
For micros where all WDT instances have the same available clock sources you can reduce this

```C
int MXC_WDT_SetClockSource(mxc_wdt_regs_t *wdt, mxc_wdt_clock_t clock_source)
{
    const uint8_t clock_source_num = 8;
    uint8_t idx = 0;
    uint8_t instance = 0;
    mxc_wdt_clock_t clock_sources[2][8] = {
        { MXC_WDT_PCLK, MXC_WDT_IPO_CLK, MXC_WDT_IBRO_CLK, MXC_WDT_INRO_CLK, MXC_WDT_ERTCO_CLK,
          MXC_WDT_EXT_CLK, MXC_WDT_ERFO_CLK, 0xFF },
        { MXC_WDT_PCLK, MXC_WDT_IPO_CLK, MXC_WDT_IBRO_CLK, MXC_WDT_INRO_CLK, MXC_WDT_ERTCO_CLK,
          MXC_WDT_EXT_CLK, MXC_WDT_ERFO_CLK, 0xFF }
    };

    if (wdt == MXC_WDT0) {
        instance = 0;
    } else if (wdt == MXC_WDT1) {
        instance = 1;
    } else {
        return E_BAD_PARAM;
    }

    for (idx = 0; idx < clock_source_num; idx++) {
        if (clock_sources[instance][idx] == clock_source) {
            break;
        }
    }

    if (idx == clock_source_num) {
        return E_BAD_PARAM;
    }

    MXC_WDT_RevB_SetClockSource((mxc_wdt_revb_regs_t *)wdt, idx);

    return E_NO_ERROR;
}
```

to this.  It will help reduce unnecessary code size.  

I'd also suggest adding a note to the `mxc_wdt_clock_t` definition that the ordering of the enum should match the register field exactly.

```C
int MXC_WDT_SetClockSource(mxc_wdt_regs_t *wdt, mxc_wdt_clock_t clock_source)
{
    MXC_WDT_RevB_SetClockSource((mxc_wdt_revb_regs_t *)wdt, (int)clock_source);

    return E_NO_ERROR;
}
```

Reply:
Updated code for ME12, ME15, ME21 and AI87.